### PR TITLE
[MONIT-28156] OpenTelemetry Metrics ingestion

### PR DESCRIPTION
**Pull request by oppegard**
### Description

This feature enables the proxy to receive OpenTelemetry Metric protobufs over the existing gRPC and HTTP listeners introduced in #656.

### Testing Notes

TBD – discuss with proxy team about available options.

### CODE REVIEW

File: proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpHttpHandler.java [hunk 1]
```diff
@@ -129,18 +139,26 @@ protected void handleHttpMessage(ChannelHandlerContext ctx, FullHttpRequest requ
...
         default:
-          String msg = "Unknown OTLP endpoint " + uri.getPath();
-          logWarning("WF-300: " + msg, null, ctx);
-          HttpResponse response = makeErrorResponse(Code.NOT_FOUND, msg);
-          writeHttpResponse(ctx, response, request);
+          /*
+          We use HTTP 200 for success and HTTP 400 for errors, mirroring what we found in
+          OTel Collector's OTLP Receiver code.
+         */
+          // TODO report internal metrics
```
Comment (joannak-vmware) [inline proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpHttpHandler.java:160]: Has this `TODO` comment been addressed?

File: proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpMetricsUtils.java [hunk 2]
```diff
@@ -0,0 +1,580 @@
...
+                                       @Nullable Supplier<ReportableEntityPreprocessor> preprocessorSupplier,
+                                       String defaultSource,
+                                       boolean includeResourceAttrsForMetrics) {
+    ReportableEntityPreprocessor preprocessor = null;
+    if (preprocessorSupplier != null) {
+      preprocessor = preprocessorSupplier.get();
+    }
+
+    for (ReportPoint point : fromOtlpRequest(request, preprocessor, defaultSource, includeResourceAttrsForMetrics)) {
+      // TODO: handle sampler
```
Comment (joannak-vmware) [inline proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpMetricsUtils.java:68]: Another `todo` comment, does this need to be addressed in this PR?

Comment (keep94) [reply proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpMetricsUtils.java:68]: Removed as there is no sampler for metrics/points.

File: proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpHttpHandler.java (hunk 1 above)
Comment (keep94) [reply proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpHttpHandler.java:160]: Are there existing templates of internal metrics that other listeners report (e.g the listener on port 2878)? We don't want to invent new metrics that may not have value.

Comment (joannak-vmware) [reply proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpHttpHandler.java:160]: These are the existing internal metrics we currently have for proxy: https://docs.wavefront.com/monitoring_proxies.html#monitoring-proxy-health-with-the-usage-dashboard. We also have internal metrics for errors in system/CollectorApiServer.

Comment (keep94) [reply proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpHttpHandler.java]: Thanks for the info Joanna. Looking at other parts of the WF Proxy code, it appears that updating internal metrics upon sending 400 http status is not done. After consulting with my team, we feel that we don't want to add special internal metrics for 400 statuses just for OTEL.  If later we want to start an internal metric for 400 status codes, it should be a separate story and apply to all intake formats and not be a part of this PR.

**Review by joannak-vmware — APPROVED** [review]
LGTM

**Review by sbhakta-vmware — APPROVED** [review]
lgtm. tested the functionality with @deosu

**Pull request merged**